src/python/tests/core/api/test_browser_assets.py

```diff
@@ -0,0 +1,635 @@
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?seconds={}'.format(test_data_asset_code, sensor_code_1, 180))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['average'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['max'] == self.test_data_x_val_list[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546, FOGL-547")
+    async def test_get_asset_sensor_readings_stats_q_min(self):
+        """
+        Verify that if more than 20 readings, summary of only last n min readings for a sensor value are returned when
+        minutes is passed as query parameter
+        http://localhost:8082/foglamp/asset/TESTAPI/x?minutes=20
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?minutes={}'.format(test_data_asset_code, sensor_code_1, 20))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == min(self.test_data_x_val_list[-2:])
+        assert retval[sensor_code_1]['average'] == \
+               sum(self.test_data_x_val_list[-2:]) / len(self.test_data_x_val_list[-2:])
+        assert retval[sensor_code_1]['max'] == max(self.test_data_x_val_list[-2:])
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_stats_q_hrs(self):
+        """
+        Verify that if more than 20 readings, summary of only last n hrs readings for a sensor value are returned when
+        hours is passed as query parameter
+        http://localhost:8082/foglamp/asset/TESTAPI/x?hours=2
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?hours={}'.format(test_data_asset_code, sensor_code_1, 2))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == min(self.test_data_x_val_list[-3:])
+        assert retval[sensor_code_1]['average'] == \
+               sum(self.test_data_x_val_list[-3:]) / len(self.test_data_x_val_list[-3:])
+        assert retval[sensor_code_1]['max'] == max(self.test_data_x_val_list[-3:])
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_stats_q_time_complex(self):
+        """
+        Verify that if a combination of hrs, min, sec is used, shortest period will apply for sensor reading
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary?hours=20&minutes=20&seconds=180&limit=20
```

Comment:
summary of last 20 records? Or summary of last 20 records in period `hours=20&minutes=20&seconds=180` logic for time is shortest and for LIMIT is AND?